src/Symfony/Component/Translation/Extractor/PhpExtractor.php

```diff
@@ -329,6 +329,10 @@ protected function canBeExtracted(string $file)
      */
     protected function extractFromDirectory($directory)
     {
+        if (!class_exists(Finder::class)) {
+            throw new \LogicException('You cannot use "Symfony\Component\Finder\Finder" as the "symfony/finder" package is not installed. Try running "composer require symfony/finder".');
```

Comment:
this should rather say that we cannot use the PhpExtractor IMO, as this is what the dev tried to use.

Reply:
@stof When you need to use `PhpExtractor`, seeing a message "You can't use `PhpExtractor`" is not much helpful (you already know it from that you see an error). You need to know what to do – thus this sentence. If you look to the related issue, you see that someone pointed me that this message is already used throughout Symfony.

Reply:
Well, the message should say `You cannot use ...\PhpExtractor as the "symfony/finder" package is not installed. Try running "composer require symfony/finder".`. What the user is trying to use is not the Finder class, so telling them they cannot use the Finder class would confuse them about why we tell them they cannot use it.
See the search link given by @ro0NL in the issue. My suggestion is consistent with all other usages. We always tell you that you cannot use the current class, not the dependent class (that Symfony tries to use but cannot find)

Reply:
I see, sorry, I was stuck inside the problem and didn't see the context already.

Can you also please answer me [my question](https://github.com/symfony/symfony/pull/42808#issuecomment-909148560) about branches so that I could fix the message in already correct branch? Thank you.

Reply:
Fixed in 78341cd933aa192a51d07e29621efb5b2cff3afe. By the way wouldn't it be better to use `self::class` instead of manually typing whole class name?